src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -1633,17 +1661,166 @@ public void checkIfStatusDateChangesWhenItemStatusUpdated()
+
+    itemsClient.replace(id, itemWithUpdatedStatusDate);
+    JsonObject itemWithUpdatedStatusDateResponse = itemsClient.getById(id).getJson();
+
+    String oldStatusDate = updatedStatus.getString("date");
+    String newStatusDate = itemWithUpdatedStatusDateResponse.getJsonObject("status").getString("date");
+
+    assertThat(newStatusDate, is(oldStatusDate));
+  }
+
+  @Test
+  public void cannotSetStatusDate() throws Exception {
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+    UUID id = UUID.randomUUID();
+    JsonObject itemToCreate = smallAngryPlanet(id, holdingsRecordId);
+    createItem(itemToCreate);
+
+    JsonObject createdItem = getById(id).getJson();
+    JsonObject initialStatus = createdItem.getJsonObject("status");
+
+    assertThat(initialStatus.getString("name"), is(Status.Name.AVAILABLE.value()));
+    assertThat(initialStatus.getString("date"), nullValue());
+
+    itemsClient.replace(id,
+      createdItem.copy()
+        .put("status", initialStatus.put("date", DateTime.now().toString()))
+    );
+
+    Response updatedItemResponse = itemsClient.getById(id);
+    JsonObject updatedStatus = updatedItemResponse.getJson().getJsonObject("status");
+
+    assertThat(updatedStatus.getString("name"), is(Status.Name.AVAILABLE.value()));
+    assertThat(updatedStatus.getString("date"), nullValue());
+  }
+
+  @Test
+  public void statusUpdatedDateRemainsAfterUpdate() throws Exception {
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+    UUID id = UUID.randomUUID();
+    JsonObject itemToCreate = smallAngryPlanet(id, holdingsRecordId);
+    createItem(itemToCreate);
+
+    JsonObject itemWithUpdatedStatus = getById(id).getJson().copy()
+      .put("status", new JsonObject().put("name", Status.Name.CHECKED_OUT.value()));
+
+    itemsClient.replace(id, itemWithUpdatedStatus);
+
+    Response updatedItemResponse = itemsClient.getById(id);
+    JsonObject updatedStatus = updatedItemResponse.getJson().getJsonObject("status");
+
+    assertThat(updatedStatus.getString("name"), is(Status.Name.CHECKED_OUT.value()));
+    assertThat(updatedStatus.getString("date"), withinSecondsBeforeNowAsString(seconds(2)));
+
+    JsonObject itemWithUpdatedCallNumber = updatedItemResponse.getJson().copy()
+      .put("itemLevelCallNumber", "newItemLevelCallNumber");
+
+    itemsClient.replace(id, itemWithUpdatedCallNumber);
+    JsonObject itemWithUpdatedCallNumberResponse = itemsClient.getById(id).getJson();
+
+    assertThat(itemWithUpdatedCallNumberResponse.getString("itemLevelCallNumber"),
+      is("newItemLevelCallNumber"));
+
+    String oldStatusDate = updatedStatus.getString("date");
+    String newStatusDate = itemWithUpdatedCallNumberResponse.getJsonObject("status").getString("date");
+
+    assertThat(oldStatusDate, is(newStatusDate));
+  }
+
+  @Test
+  public void statusUpdatedDateIsNullOnSubsequentUpdates() throws Exception {
```

Comment:
When should status updated date be null?

Reply:
Yes, correct.

Reply:
Ok, thanks